pkg/kubecost/asset.go

```diff
@@ -2902,23 +2903,26 @@ type Diff[T any] struct {
 	Kind   DiffKind
 }
 
-// DiffAsset takes two AssetSets and returns a slice of Diffs by checking
-// the keys of each AssetSet. If a key is not found, a Diff is generated
-// and added to the slice.
-func DiffAsset(before, after *AssetSet) []Diff[Asset] {
-	changedItems := []Diff[Asset]{}
+// DiffAsset takes two AssetSets and returns a map of keys to Diffs by checking
+// the keys of each AssetSet. If a key is not found or is found with a different total cost,
+// a Diff is generated and added to the map.
+func DiffAsset(before, after *AssetSet) map[string]Diff[Asset] {
+	changedItems := map[string]Diff[Asset]{}
 
 	for assetKey1, asset1 := range before.assets {
-		if _, ok := after.assets[assetKey1]; !ok {
+		if asset2, ok := after.assets[assetKey1]; !ok {
 			d := Diff[Asset]{asset1, DiffRemoved}
-			changedItems = append(changedItems, d)
+			changedItems[assetKey1] = d
+		} else if asset1.TotalCost() != asset2.TotalCost() {
+			d := Diff[Asset]{asset1, DiffChanged}
+			changedItems[assetKey1] = d
```

Comment:
Since we have two assets to compare now, what do you think about the `Diff` type including a before and after for `T`, so users can see why the change happened? Like so:

```go
type Diff[T any] struct {
    Before T
    After T
    Kind DiffKind
}
```

We have this opportunity because we're messing with the API schema a bit anyway.

Reply:
I agree that this is necessary, will work on getting this added